cset-workflow/meta/rose-meta.conf

```diff
@@ -115,697 +232,1353 @@ sort-key=met1
 type=quoted
 compulsory=true
+sort-key=a2
+
+[template variables=m01_date_type]
+ns=Models and Cases/Model 01
+title=Date placeholder type
+description=Type of date templated into the data path.
+help=The type of date templated into the data path. Affects interpretation of
+    strftime placeholders in the data path.
+
+    Forecast initiation time, AKA data time, is the time of the validity time of
+    the first data point. For realtime forecasts this is approximately when the
+    forecast was started.
 
-[template variables=CSET_INCREMENTAL_OUTPUT]
-ns=Data and Cycling
-description=Write partial output before cycle end.
-help=Whether to write partial output plots before the end of the workflow.
-    Enabling this will slow the workflow, but allows for inspecting output before
-    the workflow is complete.
+    Validity time is when the data is predicting for.
 
-    Only recommended for long running (multi-hour+) workflows.
+    Forecast lead time is how far from the forecast initiation time the data is.
+values="initiation", "validity", "lead"
+value-titles=Forecast Initiation Time, Validity Time, Forecast Lead Time
+compulsory=true
+trigger=template variables=m01_data_period: "validity", "lead";
+sort-key=b1
+
+[template variables=m01_data_period]
+ns=Models and Cases/Model 01
+title=Data period
+description=The period of the underlying data
+help=The period of the input data in each file as an ISO 8601 duration. This
+    is used to fetch files at regular validity or lead times.
+    E.g. for hourly data: PT1H
+type=quoted
+compulsory=true
+sort-key=b2
+
+[template variables=m01_preprocessing]
+ns=Models and Cases/Model 01
+title=Preprocess
+description=Preprocess all of the model data.
+help=Whether to preprocess all of the model data. This is useful for applying
+    any necessary transformations to the data before it is used in the workflow,
+    such as removing boundary regions.
 type=python_boolean
-trigger=template variables=CSET_INCREMENTAL_OUTPUT_PERIOD: True;
 compulsory=true
-sort-key=incr_out1
+trigger=template variables=m01_preprocessing_recipe: True;
+sort-key=c1
+
+[template variables=m01_preprocessing_recipe]
+ns=Models and Cases/Model 01
+title=Preprocessing recipe
+description=The preprocessing recipe to use. NOTE: Not yet implemented.
+help=The preprocessing recipe to use.
+type=quoted
+compulsory=true
+sort-key=c2
 
-[template variables=CSET_INCREMENTAL_OUTPUT_PERIOD]
-ns=Data and Cycling
-description=Period of updating output plots.
-help=ISO 8601 duration for the length of time between updating the output plots.
-    Smaller values will frequently update the output plots, but use more compute.
 
-    To avoid races, the total length of the cycles must be an integer multiple
-    of the CSET_CYCLE_PERIOD.
+# Model 02
+[template variables=m02_name]
+ns=Models and Cases/Model 02
+title=Model name
+description=A friendly name for the model.
+help=A recognisable name for this particular model. This is carried through to
```

Comment:
This name ends up in the output HTML, so I think we should support all kinds of weird characters, etc. While we escape it before templating into the HTML, I think it is possible to inject code into the cylc workflow with some carefully crafted values due to the limits of escaping here, e.g:

```cylc
' """
# Cylc code, but no Jinja, here.
_ = """
```

To avoid this I probably need to wrap every variable substitution in `shlex.quote(shlex.quote(MY_VAR))` to deal with the double shell interpretation that cylc does.

It would get a bit ungainly if more than a sentence though, so perhaps I should mention that. Then again, the non-expanding single line text entry box might be enough suggestion for that.

Reply:
I've added some clarification that the string should be reasonable short.